src/frontend/hooks/useDeepLink.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+
+//This needs to be opened on each initial screen (aka, when we have an
+//onboarding stack, this needs to be put on the initial screen of the onboarding
+//stack as well).
+import { Linking, Platform } from "react-native";
+import { useNavigation } from "react-navigation-hooks";
+
+export function useDeepLink() {
+  const { navigate } = useNavigation();
+
+  React.useEffect(() => {
+    deepLinkSetUp();
+
+    return () => deepLinkTakeDown();
+  });
+
+  function handleUrl({ url }: { url: string | null }) {
+    if (!url) return;
+    const route = url.replace(/.*?:\/\//g, "");
+    const id = route.match(/\/([^\/]+)\/?$/)![1];
+    const routeName = route.split("/")[0];
+    // if(routeName === "projInvite")
+    // {
+    //     navigate("InviteModal", {id})
+    // }
+
+    //Placeholder
+    navigate("CategoryChooser");
+  }
+
+  function deepLinkSetUp() {
+    console.log("This is being activated");
+    if (Platform.OS === "android") {
+      Linking.addEventListener("url", handleUrl);
+
+      Linking.getInitialURL().then(url => {
+        handleUrl({ url });
+      });
+    }
+  }
+
+  function deepLinkTakeDown() {
+    Linking.removeEventListener("url", handleUrl);
+  }
+}
```

Comment:
@ErikSin wondering why not use the [React Navigation](https://reactnavigation.org/docs/4.x/deep-linking) built-in support for deep linking? It avoids us having to do our own routing (e.g. matching URLs and directing to the right screen) since react-navigation does that for us, and it also passes through the router params, which we will also need for passing the encoded project key. It will be easy then to add a path for each screen so that we can open screens that are not accessible in the app (e.g. the new screens you are creating).

FYI the adb command to test deep linking is:

```
adb shell am start -W -a android.intent.action.VIEW -d "mapeo://xxxxxxxx" com.mapeo.debug
```

To avoid using ADB also useful is https://play.google.com/store/apps/details?id=com.manoj.dlt

Reply:
Yeah I'm thinking that using React Navigation's built-in support might cause fewer headaches down the line

Reply:
Right! So the reasoning why I did this was if we move to react nav v5, its slightly different. So I wanted it to be cross compatible first, and then I think switching will be easy enough

Reply:
just created a PR to use React Navigation's built-in support for it: #658